include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -46,26 +159,36 @@ class device_policy
   public:
     using kernel_name = KernelName;
 
-    device_policy() = default;
+    device_policy() : device_policy(__internal::__get_default_queue()) {}
     template <typename OtherName>
-    device_policy(const device_policy<OtherName>& other) : q(other.queue())
-    {
-    }
-    explicit device_policy(sycl::queue q_) : q(q_) {}
-    explicit device_policy(sycl::device d_) : q(d_) {}
-    operator sycl::queue() const { return q; }
+    device_policy(const device_policy<OtherName>& other) : __qh(other.queue()) {}
+    explicit device_policy(sycl::queue q) : __qh(q) {}
+    explicit device_policy(sycl::device d) : __qh(d) {}
```

Comment:
```suggestion
    explicit device_policy(sycl::queue q) : __qh(std::move(q)) {}
    explicit device_policy(sycl::device d) : __qh(std::move(d)) {}
```

I understand that this is not a part of this PR but would you be so kind to make this improvement? :)